edu.cuny.citytech.foreachlooptolambda.ui/src/edu/cuny/citytech/foreachlooptolambda/ui/visitor/LambdaConversionVisitor.java

```diff
@@ -3,160 +3,9 @@
- 	//---
- 	public boolean visit(MethodInvocation node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(ParenthesizedExpression node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(PostfixExpression node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(PrefixExpression node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(QualifiedName node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(QualifiedType node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(ReturnStatement node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(SimpleName node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(SimpleType node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(SingleVariableDeclaration node) {
- 		return visitNode(node);
- 	}
- 	//---
- 	//Yup we're counting calls to super() as well
- 	//---
- 	public boolean visit(SuperConstructorInvocation node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(SuperFieldAccess node) {
- 		return visitNode(node);
- 	}
- 	//---
- 	//super.X() counts as well
- 	//---
- 	public boolean visit(SuperMethodInvocation node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(ThisExpression node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(TypeDeclaration node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(TypeDeclarationStatement node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(TypeLiteral node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(VariableDeclarationExpression node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(VariableDeclarationStatement node) {
- 		return visitNode(node);
- 	}
- 	public boolean visit(VariableDeclarationFragment node) {
- 		return visitNode(node);
- 	}
-
-
- 	public boolean visit(IfStatement node) {
- 		return true;
- 	}
- 
- 	public boolean visit(ForStatement node) {
-	
-		return true;
-		}
- 	
 	//visiting for-each loop 
- 	public boolean visit(EnhancedForStatement node){
- 		return super.visit(node);
- 	}
- 	
- 	public boolean visit(ThrowStatement node) {
- 		return true;
- 	}
-
- 	public boolean visit(TryStatement node) {
- 		return true;
- 	}
- 	public boolean visit(InstanceofExpression node) {
- 		return true;
- 	}
- 	
- 	public boolean visit(NullLiteral node) {
- 		return false;
- 	}
- 	public boolean visit(NumberLiteral node) {
- 		return false;
- 	}
- 	public boolean visit(PackageDeclaration node) {
- 		return false;
- 	}
- 	
- 	public boolean visit(ContinueStatement node) {
- 		return false;
- 	}
- 	public boolean visit(Modifier node) {
- 		return false;
- 	}
- 	
- 	public boolean visit(MethodRef node) {
- 		return false; // comments
- 	}
- 	public boolean visit(MethodRefParameter node) {
- 		return false; // comments
- 	}
- 	public boolean visit(MemberRef node) {
- 		return false; // comments
- 	}
- 	
+	public boolean visit(EnhancedForStatement node) {
+		return super.visit(node);
```

Comment:
You tell me. What do you see in the output? Also, if it's a hint, the body of your loop is unchanged upon each iteration. Looks like you are doing the same thing over and over. Besides, I believe that `getLength()` returns the number of characters.